completions/help

```diff
@@ -0,0 +1,17 @@
+# bash completion for help                                 -*- shell-script -*-
+
+_comp_cmd_help()
+{
+    local cur prev words cword
+    _init_completion || return
+
+    if [[ $cur == -* ]]; then
+        COMPREPLY=($(compgen -W '$(_parse_usage "$1" "-s $1")' -- "$cur"))
+        return
+    fi
+
+    local IFS=$'\t\n'
+    COMPREPLY=($(compgen -A helptopic -- "$cur"))
+    ((${#COMPREPLY[*]} != 1)) || printf -v "COMPREPLY[0]" %q "$COMPREPLY"
+}
+complete -F _comp_cmd_help help
```

Comment:
Maybe this is unrelated to this PR, but I have been wondering for a long time what would be the background of the idiom used in bash-completion, `_cmd() { ... } && complete -F _cmd cmd`. In this new `help` completion, this idiom is not used, so I am interested in its background if any. I'd like to learn it for my future contributions.

I'm not sure if it is related, but as far as I know, the function definition fails when extended glob is used without `shopt -s extglob` or there is an existing shell function of the same name and marked as readonly by `declare -fr _cmd`. Are there other cases of failing function definitions?

Reply:
That's an old practice, ISTR it's related to the function definition failing but the name ending up being used as the completion function anyway, causing `bash: completion: function [...] not found` when invoked. I don't remember the exact case that provoked sprinkling the `&&`s everywhere, so for now this is mostly cargo culted as far as I'm concerned. I wouldn't mind dropping that practice if we don't see it being useful any longer.

But in this particular case, leaving it out is an oversight, will fix.

Reply:
OK! Thank you for your explanation! As far as it is consistent over the codebase, I don't mind if it's *mostly cargo-culted*.

Reply:
b9db334eded61c1538337b8b8f35919faaa35cc4